rosidl_generator_py/CMakeLists.txt

```diff
@@ -0,0 +1,63 @@
+cmake_minimum_required(VERSION 2.8.3)
+
+project(rosidl_generator_py)
+
+find_package(ament_cmake REQUIRED)
+find_package(ament_cmake_python REQUIRED)
+
+ament_export_dependencies(rosidl_cmake)
+
+ament_python_install_package(${PROJECT_NAME})
+
+if(AMENT_ENABLE_TESTING)
+  find_package(ament_lint_auto REQUIRED)
+  ament_lint_auto_find_test_dependencies()
+
+  set(message_files
+    "msg/Bool.msg"
+    "msg/Byte.msg"
+    "msg/Char.msg"
+    "msg/Empty.msg"
+    "msg/Float32.msg"
+    "msg/Float64.msg"
+    "msg/Int16.msg"
+    "msg/Int32.msg"
+    "msg/Int64.msg"
+    "msg/Int8.msg"
+    "msg/Nested.msg"
+    "msg/Strings.msg"
+    "msg/Uint16.msg"
+    "msg/Uint32.msg"
+    "msg/Uint64.msg"
+    "msg/Uint8.msg"
+    "msg/Various.msg"
+  )
+
+  include(cmake/register_py.cmake)
+  set(rosidl_generator_py_DIR "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
+
+  rosidl_generator_py_extras(
+    "${CMAKE_CURRENT_SOURCE_DIR}/bin/rosidl_generator_py"
+    "${CMAKE_CURRENT_SOURCE_DIR}/rosidl_generator_py/__init__.py"
+    "${CMAKE_CURRENT_SOURCE_DIR}/resource"
+  )
+
+  rosidl_generate_interfaces(${PROJECT_NAME} ${message_files}
+    SKIP_INSTALL
+  )
+
+  ament_add_nose_test(test_interfaces_py "${CMAKE_CURRENT_BINARY_DIR}/test")
```

Comment:
> where is the file `${CMAKE_CURRENT_BINARY_DIR}/test` supposed to come from?

From here https://github.com/ros2/rosidl/tree/python-generator/rosidl_generator_py/test

> is `test/test_interfaces.py` not supposed to contain the tests (without any templating)?

Yes, but since I couldn't get it to be available during the build phase, I moved on to other things and get back to adding tests once I resolved the aforementioned issue.